### DISCUSSION TITLE
input template

### DISCUSSION THREAD

**Discussion by Deba57** [dtcenter/METplus#2323] [Configuration] [answered]
Hi,
I am looking for how to change input template by lead time to read all the forecast files  which look like the following :
temp.instavg.20120102_0600z.nc 
temp.instavg.20120102_1200z.nc 
temp.instavg.20120102_1800z.nc 
temp.instavg.20120103_0000z.nc 
temp.instavg.20120103_0600z.nc 
temp.instavg.20120103_1800z.nc 
temp.instavg.20120104_0000z.nc 
temp.instavg.20120103_0600z.nc 
...SNIP
temp.instavg.20120206_0000z.nc

Thanks
Deba

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @Deba57, I see you have a question about setting up a METplus use case to loop over multiple forecast lead times. Looking at the filenames you provided as an example, it looks like the timestamps include the forecast valid time but make no mention of the model initialization or lead times. So I'll make the following assumptions...

- The first file you listed (`temp.instavg.20120102_0600z.nc`) corresponds to forecast hour 0. So the model was initialized at 20120102_06.
- The last file you listed (`temp.instavg.20120206_0000z.nc`) corresponds to the last forecast hour. If I'm doing my math correctly, that timestamp would be a whopping 834 hours later than the initialization.
- Even though you don't list all forecast hours, I'll assume they occur every 6 hours.
- Since I don't know anything about the full path to these files, I'll assume that they actually exist in a directory that indicates the model initialization time (e.g. `2021010206/temp.instavg.20120102_0600z.nc`). That's a total guess, but surely there's some additional time info somewhere in the directory structure.

The METplus [Example wrapper ](https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/Example/Example.html#sphx-glr-generated-met-tool-wrapper-example-example-py) is a really great tool for testing this sort of thing out.

Here are my relevant settings for that:
```
LOOP_BY = INIT
INIT_TIME_FMT = %Y%m%d%H%M
INIT_BEG = 202101020600
INIT_END = 202101020600
INIT_INCREMENT = 06H

LEAD_SEQ = begin_end_incr(0, 834, 6)H

EXAMPLE_INPUT_DIR = /dir/containing/example/data
EXAMPLE_INPUT_TEMPLATE = {init?fmt=%Y%m%d%H}/temp.instavg.{valid?fmt=%Y%m%d_%H%M}.nc
```

With this sample [DebaExample.conf.gz](https://github.com/dtcenter/METplus/files/12456200/DebaExample.conf.gz) file, I ran:
```
gunzip DebaExample.conf.gz
mkdir -p metplus_output
ush/run_metplus.py -c DebaExample.conf
```
And note the first and last files listed in the log output:
```
08/28 15:56:52.462Z metplus.4c249231 INFO: Processing forecast lead 0 hours initialized at 2021-01-02 06Z and valid at 2021-01-02 06Z
08/28 15:56:52.463Z metplus.4c249231 INFO: Looking in input directory for file: 2021010206/temp.instavg.20210102_0600.nc
...
08/28 15:56:52.498Z metplus.4c249231 INFO: Processing forecast lead 34 days 18 hours initialized at 2021-01-02 06Z and valid at 2021-02-06 00Z
08/28 15:56:52.498Z metplus.4c249231 INFO: Looking in input directory for file: 2021010206/temp.instavg.20210206_0000.nc
```

Note that you do need to pick a [LOOP_BY](https://metplus.readthedocs.io/en/latest/Users_Guide/systemconfiguration.html#loop-by) option. Here I chose to loop by initialization time but setting `INIT_BEG` and `INIT_END` to the same time, only that single initialization time is processed.

Hope that helps clarify.

**Reply by JohnHalleyGotway** [reply]
@Deba57 wanted to follow up on this discussion from a few months ago. Can we mark it as being answered or is more discussion needed?
